task/package-operator-package/0.1/package-operator-package.yaml

```diff
@@ -0,0 +1,112 @@
+      description: Path within the check out src repo containing the package definition.
+      type: string
+    - name: DST_URL
+      description: URL where to push the generated pko package to.
+      type: string
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    env:
+      - name: SRC_REPO_URL
+        value: $(params.SRC_REPO_URL)
+      - name: SRC_REF
+        value: $(params.SRC_REF)
+      - name: SRC_PATH
+        value: $(params.SRC_PATH)
+      - name: DST_URL
+        value: $(params.DST_URL)
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+  - name: git-clone
+    image: quay.io/konflux-ci/git-clone@sha256:4e53ebd9242f05ca55bfc8d58b3363d8b9d9bc3ab439d9ab76cdbdf5b1fd42d9
+    workingDir: /var/workdir
+    script: |
+      #!/bin/bash
+      set -eu -o pipefail
+      git clone -b "$SRC_REF" "$SRC_REPO_URL" repo
+  - name: build-pkg
+    image: quay.io/redhat-services-prod/mos-lpsre-tenant/package-operator-internal/kubectl-package-internal:f02995b241a7d1a8afbc60b9dec233b03179328c7d4718c6d7360758d0b13794@sha256:f02995b241a7d1a8afbc60b9dec233b03179328c7d4718c6d7360758d0b13794
+    workingDir: /var/workdir
+    script: |
+      #!/bin/bash
+      set -eu -o pipefail
+      kubectl-package build --tag "$DST_URL" --push "repo/$SRC_PATH"
+  - name: sbom
+    image: quay.io/konflux-ci/appstudio-utils:48c311af02858e2422d6229600e9959e496ddef1@sha256:91ddd999271f65d8ec8487b10f3dd378f81aa894e11b9af4d10639fd52bba7e8
+    workingDir: /var/workdir
+    script: |
+      #!/bin/bash
+      set -eu -o pipefail
+      echo '{
+        "spdxVersion": "SPDX-2.3",
+        "dataLicense": "CC0-1.0",
+        "documentNamespace": "https://package-operator.run",
+        "SPDXID": "SPDXRef-DOCUMENT",
+        "creationInfo": {
+          "creators": ["Tool: kubectl-package"]
+        },
+        "name": "package-operator-package",
+        "packages": [
+          {
+            "SPDXID": "SPDXRef-package-operator-package",
+            "name": "package-operator-package",
+            "downloadLocation": "NOASSERTION",
+            "licenseConcluded": "NOASSERTION",
+            "supplier": "NOASSERTION",
+            "externalRefs": [
+              {
+                "referenceType": "purl",
+                "referenceCategory": "PACKAGE-MANAGER"
+              }
+            ]
+          }
+        ],
+        "relationships": [
+          {
+            "spdxElementId": "SPDXRef-DOCUMENT",
+            "relationshipType": "DESCRIBES",
+            "relatedSpdxElement": "SPDXRef-package-operator-package"
+          }
+        ]
+      }
+      ' | jq \
+        --arg datearg "$(date --iso-8601=seconds -u)" \
+        --arg srcrefarg "$SRC_REF" \
+        --arg urlarg "${SRC_REPO_URL}?ref=${SRC_REF}&path=${SRC_PATH}" \
+        '.
+        | .creationInfo.created |= $datearg
+        | .packages[0].versionInfo |= $srcrefarg
+        | .packages[0].externalRefs[0].referenceLocator = $urlarg
```

Comment:
The `$urlarg` is not a valid purl, see https://github.com/package-url/purl-spec/tree/main.

If `kubectl package` cannot generate an SBOM, it may be best to contribute this functionality to https://github.com/konflux-ci/mobster. That would improve the consistency with other Konflux SBOMs and would increase the chance that it ends up being valid.

And if we're saying that the only thing we need in the SBOM is the source repo url, then it can be a generic piece of functionality to generate an SBOM with repo information

Reply:
Thank you for the reference for purl, overlooked that. 

Currently `kubectl package` can't generate SBOMs, discussed with the team if we implement it there but decided not to. 

I would like to only specify the git repo, the checked out revision and the path within the checked out workspace, I put those together into a URL.

Will open a PR to mobster first then, thanks!